src/pages/GetStarted/CustomWorkspaceTab/index.tsx

```diff
@@ -46,111 +40,144 @@ export class CustomWorkspaceTab extends React.Component<Props, State> {
+        }
+        devfile.attributes.persistVolumes = 'false';
+        delete devfile.attributes.asyncPersist;
+        break;
+      case StorageType.async:
+        if (!devfile.attributes) {
+          devfile.attributes = {};
+        }
+        devfile.attributes.persistVolumes = 'false';
+        devfile.attributes.asyncPersist = 'true';
+        break;
     }
 
-    this.setState({ devfile });
-    this.updateEditor(devfile);
+    this.setState({ storageType, devfile });
+    this.updateEditor();
   }
 
-  private handleNewDevfile(devfileContent: string): void {
-    const devfile = load(devfileContent);
+  private handleNewDevfile(devfileContent?: che.WorkspaceDevfile): void {
+    const devfile = devfileContent ? devfileContent : this.buildInitialDevfile();
+    const { storageType, workspaceName, namespace } = this.state;
     this.setState({ devfile });
+    if (namespace) {
+      this.handleInfrastructureNamespaceChange(namespace);
+    }
+    if (workspaceName) {
+      this.handleWorkspaceNameChange(workspaceName, devfile);
+    }
+    if (storageType) {
+      this.handleStorageChange(storageType, devfile);
+    }
     this.updateEditor(devfile);
   }
 
+  private getStorageType(devfile: che.WorkspaceDevfile): StorageType {
+    let storageType: StorageType;
+    // storage type
+    if (devfile.attributes && devfile.attributes.persistVolumes === 'false') {
+      const isAsync = devfile.attributes && devfile.attributes.asyncPersist === 'true';
+      if (isAsync) {
+        storageType = StorageType.async;
+      } else {
+        storageType = StorageType.ephemeral;
+      }
+    } else {
+      storageType = StorageType.persistent;
+    }
+    return storageType;
+  }
+
   private handleDevfileChange(devfile: che.WorkspaceDevfile, isValid: boolean): void {
     if (!isValid) {
       return;
     }
-
-    // temporary storage
-    const isTemporary = !!devfile.attributes && devfile.attributes?.persistVolumes === 'false';
-    if (isTemporary !== this.state.isTemporary) {
-      this.setState({ isTemporary });
+    this.setState({ devfile });
+    const storageType = this.getStorageType(devfile);
+    if (storageType !== this.state.storageType) {
+      this.setState({ storageType });
     }
-
-    // workspace name
     const workspaceName = devfile.metadata.name || '';
+    if (workspaceName !== this.state.workspaceName) {
+      this.setState({ workspaceName });
+    }
     const generateName = devfile.metadata.generateName;
-    if (workspaceName !== this.state.workspaceName || generateName !== this.state.generateName) {
-      this.setState({
-        workspaceName,
-        generateName,
-      });
+    if (generateName !== this.state.generateName) {
+      this.setState({ generateName });
     }
   }
 
-  private updateEditor(devfile: che.WorkspaceDevfile): void {
-    this.devfileEditorRef.current?.updateContent(devfile);
+  private updateEditor(devfile?: che.WorkspaceDevfile): void {
```

Comment:
You don't have to pass a devfile to `updateEditor`. Each time this method is called the state already contains the actual devfile.

Reply:
I am using both. It is more convenient.

Reply:
This line is the only place the devfile is passed to this method: 
https://github.com/che-incubator/che-dashboard-next/pull/60/files#diff-b7189c84d088e265b9bc417459214d26R133
And few lines above component state is updated with the devfile: 
https://github.com/che-incubator/che-dashboard-next/pull/60/files#diff-b7189c84d088e265b9bc417459214d26R123

My point is to call `this.updateEditor` without parameters and get actual devfile from the state.

Reply:
OK. Done